src/zenml/config/compiler.py

```diff
@@ -643,12 +645,50 @@ def _compute_pipeline_spec(
                 "https://docs.zenml.io/user-guides/starter-guide"
             )
 
-        additional_spec_args: Dict[str, Any] = {
-            "source": pipeline.resolve(),
-            "parameters": pipeline._parameters,
-        }
+        output_specs = [
+            OutputSpec(
+                step_name=output_artifact.invocation_id,
+                output_name=output_artifact.output_name,
+            )
+            for output_artifact in pipeline._output_artifacts
+        ]
+        output_schema = compute_pipeline_output_schema(pipeline=pipeline)
+
+        return PipelineSpec(
+            steps=step_specs,
+            outputs=output_specs,
+            output_schema=output_schema,
+            source=pipeline.resolve(),
+            parameters=pipeline._parameters,
+        )
+
 
-        return PipelineSpec(steps=step_specs, **additional_spec_args)
+def compute_pipeline_output_schema(
+    pipeline: "Pipeline",
+) -> Optional[Dict[str, Any]]:
+    """Computes the pipeline output schema.
+
+    Args:
+        pipeline: The pipeline for which to compute the output schema.
+
+    Returns:
+        The pipeline output schema.
+    """
+    if not pipeline._output_artifacts:
+        return None
+
+    output_model = create_model(
+        f"{pipeline.name}_output",
+        __config__=ConfigDict(arbitrary_types_allowed=True),
+        **{
+            output_artifact.output_name: (
```

Comment:
@stefannica @safoinme How should we handle the case where a pipeline returns multiple step outputs, all of which have the same name?
- Should the pipeline output names be something like `<step-name>_<output_name>`?
- Should I just add suffixes to avoid duplicates, e.g. `output`, `output_2`, ...
- Something else?

Reply:
Did this with some added logic to guarantee it's unique. But now that I think about it, for you it's probably easier if it actually uses the real step name and output name, so you can easier match those to the actual step outputs, correct? I can also remove my `.replace("-", "_")` calls if you feel that's better.